.github/workflows/nightly_test.yml

```diff
@@ -0,0 +1,205 @@
+  # --- Prechecks and Checkouts ------------------------------------------------------------------------
+  # ----------------------------------------------------------------------------------------------------
+  Precheck:
+    uses: alcionai/corso/.github/workflows/_filechange_checker.yml@main
+
+  Checkout:
+    needs: [Precheck]
+    environment: Testing
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: src
+    steps:
+      - uses: actions/checkout@v3
+
+      # single setup and sum cache handling here.
+      # the results will cascade onto both testing and linting.
+      - name: Setup Golang with cache
+        uses: ./.github/actions/go-setup-cache
+        if: startsWith(github.ref, 'refs/tags/') || github.ref == 'refs/heads/main' || needs.precheck.outputs.websitefileschanged == 'true'
+        with:
+          go-version-file: src/go.mod
+
+  # SetM365App will decide which M365 app to use for this CI run
+  SetM365App:
+    environment: Testing
+    runs-on: ubuntu-latest
+    outputs:
+      client_app_slot: ${{ steps.roundrobin.outputs.CLIENT_APP_SLOT }}
+      client_id_env: ${{ steps.roundrobin.outputs.CLIENT_ID_ENV }}
+      client_secret_env: ${{ steps.roundrobin.outputs.CLIENT_SECRET_ENV }}
+    steps:
+      - name: Figure out which client id to use
+        id: roundrobin
+        run: |
+          slot=$((GITHUB_RUN_NUMBER % 4))
+          echo "CLIENT_APP_SLOT=$slot" >> $GITHUB_OUTPUT
+
+          case $slot in
+
+            0)
+              echo "CLIENT_ID_ENV=CLIENT_ID" >> $GITHUB_OUTPUT
+              echo "CLIENT_SECRET_ENV=CLIENT_SECRET" >> $GITHUB_OUTPUT
+              ;;
+            1)
+              echo "CLIENT_ID_ENV=CLIENT_ID_2" >> $GITHUB_OUTPUT
+              echo "CLIENT_SECRET_ENV=CLIENT_SECRET_2" >> $GITHUB_OUTPUT
+              ;;
+            2)
+              echo "CLIENT_ID_ENV=CLIENT_ID_3" >> $GITHUB_OUTPUT
+              echo "CLIENT_SECRET_ENV=CLIENT_SECRET_3" >> $GITHUB_OUTPUT
+              ;;
+            3)
+              echo "CLIENT_ID_ENV=CLIENT_ID_4" >> $GITHUB_OUTPUT
+              echo "CLIENT_SECRET_ENV=CLIENT_SECRET_4" >> $GITHUB_OUTPUT
+              ;;
+          esac
+
+  SetEnv:
+    environment: Testing
+    runs-on: ubuntu-latest
+    outputs:
+      environment: ${{ steps.environment.outputs.environment }}
+      version: ${{ steps.version.outputs.version }}
+      website-bucket: ${{ steps.website-bucket.outputs.website-bucket }}
+      website-cfid: ${{ steps.website-cfid.outputs.website-cfid }}
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Figure out environment
+        id: environment
+        run: |
+          if ${{ startsWith(github.ref, 'refs/tags/') }}; then
+            echo "environment=Production" | tee -a $GITHUB_OUTPUT
+          else
+            echo "environment=Testing" | tee -a $GITHUB_OUTPUT
+          fi
```

Comment:
Are we running nightly for tags separately? We should run this in the release job so that a release is blocked in case there is an issue.

Reply:
Also, don't think we should be using Production env at all.